src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,35 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final MachineProducerStrategy producerStrategy;
+
+    public MachineServiceImpl() {
+        producerStrategy = new MachineProducerStrategy();
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        return new ArrayList<>(producerStrategy.getMachineProducerStrategy(type).get());
```

Comment:
About not addressed - You mean, NPE is better than checking for null?(If null is passed as argument) Ok, maybe it's because of  a new concept or material in our course that's a little outdated.. I will raise this question. Fixed.
And it was like an eye opening for me...

Reply:
public List<Machine> getAll(Class<? extends Machine> type) {
        MachineProducer<? extends Machine> producer = producerStrategy
                .getMachineProducerStrategy(type);
        return new ArrayList<>(producer.get());
    }

and I leave comment about this peace of code. I can't initialize producer in one line because if I do, the line length will be over  100 characters and clean/package check will fail.